src/libfetchers/github.cc

```diff
@@ -172,20 +172,45 @@ struct GitArchiveInputScheme : InputScheme
         return input;
     }
 
-    std::optional<std::string> getAccessToken(const fetchers::Settings & settings, const std::string & host) const
+    std::optional<std::string> getAccessToken(const fetchers::Settings & settings, const std::string & host, const std::string & url) const override
     {
         auto tokens = settings.accessTokens.get();
+        std::string answer;
+        size_t answer_match_len = 0;
+        if(! url.empty()) {
+            for (auto & token : tokens) {
+                auto match_len = url.find(token.first);
+                if (match_len != std::string::npos && token.first.length() > answer_match_len) {
```

Comment:
This will make the URL `github.com/foobar` match against the token for `github.com/foo` and I'm not sure that's desirable. Maybe it should only match if the match is followed by a `/` or end-of-string. (Similar to `isInDir()`.)

Reply:
And in general the substring search is a bit iffy, since the token `github.com/foo` will also match `attacker.org/github.com/foo` will could theoretically be used to exfiltrate an access token? (If the attacker can get a user to evaluate an expression containing that URL, for instance.)

Maybe it should parse the URL and match the `authority` and `path` fields separately.

Reply:
Did it manually for now as the urls did not have schemes.